src/util/entropy_utils.js

```diff
@@ -0,0 +1,133 @@
+/* Copyright (C) 2025 NooBaa */
+'use strict';
+
+const fs = require('fs');
+const util = require('util');
+const chance = require('chance')();
+const child_process = require('child_process');
+const config = require('../../config');
+const blockutils = require('linux-blockutils');
+
+const DEVICE_TYPE_DISK = 'disk';
+
+const async_delay = util.promisify(setTimeout);
+const async_exec = util.promisify(child_process.exec);
+const get_block_info_async = util.promisify(blockutils.getBlockInfo);
+
+/**
+ * generate_entropy will create randomness by changing the MD5
+ * using information from the device (disk)
+ * it will run as long as the callback it true
+ * @param {function} loop_cond
+ */
+async function generate_entropy(loop_cond) {
+    if (process.platform !== 'linux' || process.env.container === 'docker') return;
+    while (loop_cond()) {
+        try {
+            await async_delay(1000);
```

Comment:
I forgot about this point - but we should check the loop condition again right after sleeping, because most likely we will read from the disk for no reason at all as the caller already finished initializing its randomness.
I would just add an `if (loop_cond()) break;` right after this delay.